localstack-core/localstack/services/lambda_/invocation/event_manager.py

```diff
@@ -201,19 +202,16 @@ def handle_message(self, message: dict) -> None:
         event_invoke_config = self.version_manager.function.event_invoke_configs.get(qualifier)
         runtime = None
         status = None
+        initialization_type = FunctionInitializationType.on_demand
         try:
             sqs_invocation = SQSInvocation.decode(message["Body"])
             invocation = sqs_invocation.invocation
             try:
                 invocation_result = self.version_manager.invoke(invocation=invocation)
                 function_config = self.version_manager.function_version.config
-                function_counter.labels(
-                    operation=FunctionOperation.invoke,
-                    runtime=function_config.runtime or "n/a",
-                    status=FunctionStatus.success,
-                    invocation_type=InvocationType.Event,
-                    package_type=function_config.package_type,
-                ).increment()
+                # TODO: handle initialization_type provisioned-concurrency, requires enriching invocation_result
```

Comment:
```suggestion
                status = FunctionStatus.success
                # TODO: handle initialization_type provisioned-concurrency, requires enriching invocation_result
```
we need to update the status here, otherwise it will be None in the finally block for successful invocations.

Reply:
Great catch 🕵️ Thank you very much for spotting this.

The complexity of this method is 🤯 . I introduced fail-fast asserts for `status` and `invocation_result` to spot such issues more easily.